### ISSUE TITLE
Getting auth error in emulator example app

### ISSUE THREAD

**Issue by zakton5** [awinogrodzki/next-firebase-auth-edge#35]
I am trying to set this up in my own project and connect to firebase emulators. I keep getting this error in both my project and the firebase emulator example in this repo:

`Error fetching access token: invalid_grant (Invalid grant: account not found)`

This tells me that something is wrong with my configuration but I have no idea what could be wrong at this point. I filled in my own values in the `.env` file. 

Do I need a special private key? A special admin client email?

**Comment by awinogrodzki** [comment]
Hey @zakton5!

Thanks for reporting. 

I cannot seem to be able to reproduce the error.

Just to understand correctly – you have filled .env file with credentials obtained from firebase project settings?

**Comment by zakton5** [comment]
@awinogrodzki Yes, that's correct. The credentials for a live firebase project work fine when put in the .env file. But when running the emulators locally it isn't working. I use the `--project` flag with a `demo` prefix (ie. `firebase emulators:start  --project=demo-project-name`. Could this affect it? Have you tested it with this flag by chance?

**Comment by awinogrodzki** [comment]
In `.env`, do you also use `demo-` prefix for project id variables? ie:

```
FIREBASE_PROJECT_ID=demo-project-name
NEXT_PUBLIC_FIREBASE_PROJECT_ID=demo-project-name
```

**Comment by awinogrodzki** [comment]
While we're at it, could you ensure the following to exclude most common external causes:

1. In Firebase console, go to Authentication > Sign-in method and make sure "Anonymous" is enabled
2. run `firebase login --reauth` to reauthenticate locally
3. There can be many causes for `invalid_grant`, but it's definitely out of ordinary. Some users report that generating new service account credentials have helped 

Even in emulator mode, library needs actual service account credentials to sign tokens with x509 certificate fetched from Google, a step that does not currently support emulation.

**Comment by zakton5** [comment]
@awinogrodzki so could I theoretically use my live development project credentials but have it work in the emulator? Those are considered different projects which I would think would cause the invalid_grant error.

I don't want to enable anonymous authentication for my project so that isn't really an option.

**Comment by zakton5** [comment]
I'll try reauthenticating after the weekend.

**Comment by awinogrodzki** [comment]
@zakton5 sure, I asked you to enable anonymous authentication, because you mentioned you are using the code from emulator example, which is using anonymous authentication :-) If it's not the case then please don't bother.



> @awinogrodzki so could I theoretically use my live development project credentials but have it work in the emulator? Those are considered different projects which I would think would cause the invalid_grant error.

I would recommend creating another project in firebase, just for local development. At least for the time being. My goal is to allow user to work with emulators without any additional configuration, but it'll require some tweaks here and there 🙏

**Comment by zakton5** [comment]
@awinogrodzki I tried using my development account credentials in the .env with my local emulators and everything worked fine. This information that you provided is important:

> Even in emulator mode, library needs actual service account credentials to sign tokens with x509 certificate fetched from Google, a step that does not currently support emulation.

Would you mind adding this to the emulator section of the README to prevent confusion, at least until this package works without configuration for emulators?
